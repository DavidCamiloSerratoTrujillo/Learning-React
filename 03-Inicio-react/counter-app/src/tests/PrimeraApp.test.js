import PrimeraApp from "../PrimeraApp";
import{shallow} from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom'
describe('Pruebas en <PrimeraApp>',()=>{

    test('debe demostrar <PrimeraApp/> correctamente',()=>{
       
        const saludo = 'Hola, soy goku';
        const wrapper = shallow(<PrimeraApp saludo = {saludo}/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar el subtitulo enviado',()=>{
        const saludo = 'Hola, soy goku';
        const subtitulo = 'Soy un subtitulo'
        const wrapper = shallow(
            <PrimeraApp 
                saludo = {saludo}
                subtitulo = {subtitulo}
            />);

        const textoParrafo = wrapper.find('h2').text();    
        
        expect( textoParrafo).toBe( subtitulo );
    });
});