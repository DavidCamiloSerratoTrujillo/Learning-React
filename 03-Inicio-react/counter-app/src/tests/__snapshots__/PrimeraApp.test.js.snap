// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp> debe demostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy goku
  </h1>
  <h2>
    Soy un subtitulo
  </h2>
</Fragment>
`;
